### PR TITLE
Implement custom origin bootstrap iframes for ads (and other 3p embeds).

### LINKED ISSUE #783: Allow serving ads and other 3p embeds from custom origins
Useful for access to 1p cookies from origin sites among other things.

### PR DESCRIPTION
**Pull request by cramforce**
Fixes #783

### CODE REVIEW

File: src/3p-frame.js [hunk 1]
```diff
@@ -17,14 +17,15 @@
...
 import {documentInfoFor} from './document-info';
 import {getMode} from './mode';
 import {dashToCamelCase} from './string';
-import {parseUrl} from './url';
+import {parseUrl, assertHttpsUrl} from './url';
 
 
 /** @type {!Object<string,number>} Number of 3p frames on the for that type. */
-var count = {};
+const count = {};
```
Comment (sriramkrish85) [inline src/3p-frame.js]: const COUNT?

Comment (cramforce) [inline src/3p-frame.js]: We'll need to discuss that rule at some point. I think it is weird and ugly here.

Comment (erwinmombay) [inline src/3p-frame.js]: use `Object.create(null)`?

Comment (cramforce) [inline src/3p-frame.js]: This was just a drive by change. I'm not doing anything with this var here.

This is fine here as long as we don't want to allow toString as a valid 3p embed type :)

**Pull request merged**